sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_models.py

```diff
@@ -658,6 +672,128 @@ def __repr__(self):
         )[:1024]
 
 
+class AspectSentiment(DictMixin):
+    """AspectSentiment contains the related opinions, predicted sentiment,
+    confidence scores and other information about an aspect of a product.
+    An aspect of a product/service is a key component of that product/service.
+    For example in "The food at Hotel Foo is good", "food" is an aspect of
+    "Hotel Foo".
+
+    :ivar str text: The aspect text.
+    :ivar str sentiment: The predicted Sentiment for the aspect. Possible values
+        include 'positive', 'mixed', and 'negative'.
+    :ivar confidence_scores: The sentiment confidence score between 0
+        and 1 for the aspect for 'positive' and 'negative' labels. It's score
+        for 'neutral' will always be 0
+    :vartype confidence_scores:
+        ~azure.ai.textanalytics.SentimentConfidenceScores
+    :ivar opinions: All of the opinions related to this aspect.
+    :vartype opinions: list[~azure.ai.textanalytics.OpinionSentiment]
+    :ivar int offset: The aspect offset from the start of the sentence.
+    :ivar int length: The length of the aspect.
+    """
+
+    def __init__(self, **kwargs):
+        self.text = kwargs.get("text", None)
+        self.sentiment = kwargs.get("sentiment", None)
+        self.confidence_scores = kwargs.get("confidence_scores", None)
+        self.opinions = kwargs.get("opinions", None)
+        self.offset = kwargs.get("offset", None)
+        self.length = kwargs.get("length", None)
+
+    @staticmethod
+    def _get_opinions(relations, results):
+        if not relations:
+            return []
+        opinion_relations = [r.ref for r in relations if r.relation_type == "opinion"]
+        opinions = []
+        for opinion_relation in opinion_relations:
+            nums = [int(s) for s in re.findall(r"\d+", opinion_relation)]
```

Comment:
This is technically incorrect parsing of the json pointer (it doesn't take escaping into account). This may or may not be an actual issue (I don't know of any of the keys can have a `/` or `~` in them). But either way, I would suggest breaking the parsing of jsonpointers out to a separate function that can be tested (and reused).

Reply:
In this case, the service will always returns something along the lines of `"#/documents/0/sentences/0/opinions/0"` as the json pointer. I've taken out the json pointer code though, and have added a unittest for it